src/sentry/ingest/billing_metrics_consumer.py

```diff
@@ -0,0 +1,126 @@
+from datetime import datetime
+from typing import Callable, Dict, Mapping, Optional, Sequence, Union
+
+from arroyo import Topic
+from arroyo.backends.kafka import KafkaConsumer, KafkaPayload
+from arroyo.backends.kafka.configuration import build_kafka_consumer_configuration
+from arroyo.processing import StreamProcessor
+from arroyo.processing.strategies import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.types import Message, Partition, Position
+from django.conf import settings
+
+from sentry.constants import DataCategory
+from sentry.sentry_metrics.indexer.strings import TRANSACTION_METRICS_NAMES
+from sentry.utils import json
+from sentry.utils.outcomes import Outcome, track_outcome
+
+
+def get_metrics_billing_consumer(
+    topic: str,
+    group_id: str,
+    auto_offset_reset: str,
+    force_topic: Union[str, None],
+    force_cluster: Union[str, None],
+    **options,
+) -> StreamProcessor[KafkaPayload]:
+    bootstrap_servers = _get_bootstrap_servers(topic, force_topic, force_cluster)
+
+    return StreamProcessor(
+        consumer=KafkaConsumer(
+            build_kafka_consumer_configuration(
+                default_config={},
+                group_id=group_id,
+                auto_offset_reset=auto_offset_reset,
+                bootstrap_servers=bootstrap_servers,
```

Comment:
@lynnagara I'm not sure this is the right way to do this. If this is not the right way, let me know what the best way is.